### PR TITLE
Fix the metric for classification module in  python example

### PR DESCRIPTION
**Pull request by Green-16**
I think  use rmse for classification module is inappropriate.   So  I use roc_auc_score replace mean_squared_error.
the reason of this update is show in the below link. https://github.com/microsoft/LightGBM/issues/5397#issuecomment-1226993663

@jameslamb

### CODE REVIEW

File: examples/python-guide/advanced_example.py [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 import numpy as np
 import pandas as pd
-from sklearn.metrics import mean_squared_error
+from sklearn.metrics import mean_squared_error, roc_auc_score
```
Comment (StrikerRUS) [inline examples/python-guide/advanced_example.py:9]:
```suggestion
from sklearn.metrics import roc_auc_score
```

File: examples/python-guide/advanced_example.py [hunk 2]
```diff
@@ -84,8 +84,8 @@
 # can only predict with the best iteration (or the saving iteration)
 y_pred = bst.predict(X_test)
 # eval with loaded model
-rmse_loaded_model = mean_squared_error(y_test, y_pred) ** 0.5
-print(f"The RMSE of loaded model's prediction is: {rmse_loaded_model}")
+auc_loaded_model = roc_auc_score(y_test, y_pred)
+print(f"The auc of loaded model's prediction is: {auc_loaded_model}")
```
Comment (StrikerRUS) [inline examples/python-guide/advanced_example.py:88]:
```suggestion
print(f"The ROC AUC of loaded model's prediction is: {auc_loaded_model}")
```

File: examples/python-guide/advanced_example.py [hunk 3]
```diff
@@ -97,8 +97,8 @@
 # can predict with any iteration when loaded in pickle way
 y_pred = pkl_bst.predict(X_test, num_iteration=7)
 # eval with loaded model
-rmse_pickled_model = mean_squared_error(y_test, y_pred) ** 0.5
-print(f"The RMSE of pickled model's prediction is: {rmse_pickled_model}")
+auc_pickled_model = roc_auc_score(y_test, y_pred)
+print(f"The auc of pickled model's prediction is: {auc_loaded_model}")
```
Comment (StrikerRUS) [inline examples/python-guide/advanced_example.py:101]:
```suggestion
print(f"The ROC AUC of pickled model's prediction is: {auc_pickled_model}")
```

**Review by StrikerRUS — CHANGES_REQUESTED** [review]

File: examples/python-guide/advanced_example.py [hunk 4]
```diff
@@ -210,6 +210,7 @@ def callback(env):
         if env.iteration - env.begin_iteration == 5:
             print('Add a new valid dataset at iteration 5...')
             env.model.add_valid(lgb_eval_new, 'new_valid')
+
```
Comment (jameslamb) [inline examples/python-guide/advanced_example.py:213]: can you please remove this unrelated whitespace change?

Comment (Green-16) [reply examples/python-guide/advanced_example.py:213]: ok， when I use  `Reformat code` command in pycharm, the new blank line will add. Anyway  I have remove it.

File: examples/python-guide/advanced_example.py (hunk 3 above)
Comment (Green-16) [reply examples/python-guide/advanced_example.py]: done

**Review by jameslamb — APPROVED** [review]
Looks good to me, thanks!

In the future when you make contributions here (or in any open source project on GitHub), I strongly recommend creating a branch on your fork instead of directly pushing to `main` / `master`.

Once this pull request is merged, you'll need to either delete your fork or do some manual rebasing and force pushing on `master` in your fork, since we squash all commits from PRs into a single commit when merging.

**Review by StrikerRUS — APPROVED** [review]
LGTM, thanks for your contribution!
